permissions/perUserMfaState/grantPermission.ps1

```diff
@@ -0,0 +1,322 @@
+    #endregion Get current PerUserMfaState
+
+    #region Calulate action
+    $actionMessage = "calculating action"
+    
+    # Remove spaces
+    
+    if ($currentPerUserMfaState -eq 'disabled') {
+        $actionPerUserMfaState = "Update"
+    }
+    else {
+        $actionPerUserMfaState = "NoChanges"
+    }
+    
+    #endregion Calulate action
+
+    #region Process
+    switch ($actionPerUserMfaState) {
+        "Update" {
+            #region Update PerUserMfaState
+            # Microsoft docs: https://learn.microsoft.com/nl-nl/graph/api/PerUserMfaState-update?view=graph-rest-1.0&tabs=http
+            $actionMessage = "setting perUserMfaState to [$($state)] for account"
+            $baseUri = "https://graph.microsoft.com/"
+
+            $updatePerUserMfaStateBody = @{
+                "perUserMfaState" = $state
+            }
+
+            $updatePerUserMfaStateSplatParams = @{
+                Uri         = "$($baseUri)/beta/users/$($actionContext.References.Account)/authentication/requirements"
+                Method      = "PATCH"
+                Body        = ($updatePerUserMfaStateBody | ConvertTo-Json -Depth 10)
+                Verbose     = $false
+                ErrorAction = "Stop"
+            }
+
+            Write-Verbose "SplatParams: $($updatePerUserMfaStateSplatParams | ConvertTo-Json)"
+
+            if (-Not($actionContext.DryRun -eq $true)) {
+                # Add Headers after printing splat
+                $updatePerUserMfaStateSplatParams['Headers'] = $headers
+
+                $updatedPerUserMfaState = Invoke-RestMethod @updatePerUserMfaStateSplatParams
+
+                $outputContext.AuditLogs.Add([PSCustomObject]@{
+                        # Action  = "" # Optional
+                        Message = "$state perUserMfaState [$($actionContext.References.Permission.Name)] for account with AccountReference: $($actionContext.References.Account | ConvertTo-Json). Old value: [$($currentPerUserMfaState)]. New value: [$($state)]."
+                        IsError = $false
+                    })
+            }
+            else {
+                Write-Warning "DryRun: Would set perUserMfaState to [$($state)] for account with AccountReference: $($actionContext.References.Account | ConvertTo-Json). Old value: [$($currentPerUserMfaState)]. New value: [$($state)]."
+            }
+            #endregion Update PerUserMfaState
+
+            break
+        }
+
+        "NoChanges" {
+            #region No changes
+            $actionMessage = "skipping setting perUserMfaState to [$($state)] for account"
+
+            $outputContext.AuditLogs.Add([PSCustomObject]@{
+                    # Action  = "" # Optional
+                    Message = "Skipped setting perUserMfaState to [$($state)] for account with AccountReference: $($actionContext.References.Account | ConvertTo-Json). Old value: [$($currentPerUserMfaState)]. New value: [$($state)]. Reason: No changes."
```

Comment:
Old value en new value lijkt me niet nodig als er geen changes zijn. Beter iets melden in de trend van
'Skipped setting perUserMfaState as state is already set to the desired state' of 'Skipped setting perUserMfaState as state is already set $state'

Reply:
aangepast